toonz/sources/toonz/xsheetdragtool.cpp

```diff
@@ -1595,22 +1597,55 @@ class ColumnMoveDragTool final : public XsheetGUI::DragTool {
     if (indices.empty()) return;
     m_firstCol = m_lastCol = *indices.begin();
     assert(m_firstCol >= 0);
-    m_offset = m_firstCol - col;
+    m_origOffset = m_offset = m_firstCol - col;
     assert(m_lastCol == *indices.begin());
     getViewer()->update();
   }
   void onDrag(const CellPosition &pos) override {
     int col                     = pos.layer();
     TColumnSelection *selection = getViewer()->getColumnSelection();
-    std::set<int> indices       = selection->getIndices();
+    TApp *app                   = TApp::instance();
+    TXsheet *xsh                = app->getCurrentXsheet()->getXsheet();
+
+    std::set<int> indices = selection->getIndices();
     if (indices.empty()) return;
 
     assert(m_lastCol == *indices.begin());
 
+    int origCol      = col;
     if (col < 0) col = 0;
-    if (col == m_lastCol) return;
-    int dCol  = col - m_lastCol;
-    m_lastCol = col;
+    int dCol         = col - (m_lastCol - m_offset);
+
+    if (!getViewer()->orientation()->isVerticalTimeline()) {
+      if (origCol < 0) dCol += origCol;
+      std::set<int> ii;
+      int newBegin = *indices.begin() + dCol;
+      if (newBegin < 0) {
+        newBegin *= -1;
+        for (int x = 0; x < newBegin; x++) ii.insert(x);
+        ColumnCmd::insertEmptyColumns(ii);
+        selection->selectNone();
+        for (std::set<int>::const_iterator it = indices.begin();
+             it != indices.end(); it++)
+          selection->selectColumn(*it + newBegin, true);
+        indices = selection->getIndices();
+        col     = origCol + newBegin;
+        m_lastCol += newBegin;
+        m_firstCol += newBegin;
+        int currentIndx = app->getCurrentColumn()->getColumnIndex();
+        app->getCurrentColumn()->setColumnIndex(currentIndx + newBegin);
+      } else {
+        int currEnd = xsh->getColumnCount() - 1;
+        int newEnd  = *indices.rbegin() + dCol;
+        if (newEnd > currEnd) {
+          for (int x = currEnd + 1; x <= newEnd; x++) ii.insert(x);
+          ColumnCmd::insertEmptyColumns(ii);
+        }
+      }
+    }
```

Comment:
For now, OT crashes by doing the following operations;
1. Select multiple columns (say Col1, 2 and 3) in Xsheet view.
1. Grab a handle of Col3 and drag to the left direction.

Inserting the following lines seems to avoid crash:
```
@@ -1642,6 +1645,11 @@ public:
           ColumnCmd::insertEmptyColumns(ii);
         }
       }
+    } else {  // xsheet view
+      if (col < -m_offset) {
+        col  = -m_offset;
+        dCol = -m_lastCol;
+      }
     }
 
     if (col == (m_lastCol - m_offset)) return;
```

Reply:
Fixed.  I simply blocked the action if the 1st column number in the selection is < 0 after applying movement offset.